onnxruntime/core/graph/graph_flatbuffers_utils.cc

```diff
@@ -0,0 +1,406 @@
+    case fbs::AttributeType_INTS: {
+      GET_DATA_VEC(int64_t, ints_vec_, attr_proto.ints());
+      auto ints = builder.CreateVector(ints_vec_);
+      GET_FBS_ATTR(builder, type, ints, ints);
+    } break;
+    case fbs::AttributeType_STRINGS: {
+      GET_DATA_VEC(std::string, strings_vec_, attr_proto.strings());
+      auto strings = builder.CreateVectorOfStrings(strings_vec_);
+      GET_FBS_ATTR(builder, type, strings, strings);
+    } break;
+    case fbs::AttributeType_TENSORS: {
+      std::vector<flatbuffers::Offset<fbs::Tensor>> fbs_tensors_vec;
+      fbs_tensors_vec.reserve(attr_proto.tensors().size());
+      for (const auto& tensor : attr_proto.tensors()) {
+        flatbuffers::Offset<fbs::Tensor> fbs_tensor;
+        ORT_RETURN_IF_ERROR(
+            experimental::utils::SaveInitializerOrtFormat(builder, tensor, fbs_tensor));
+        fbs_tensors_vec.push_back(fbs_tensor);
+      }
+      auto tensors = builder.CreateVector(fbs_tensors_vec);
+      GET_FBS_ATTR(builder, type, tensors, tensors);
+    } break;
+    default:
+      break;
+  }
+
+  return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "GetAttributeOrtFormat - Unsupported type: ", type);
+}
+
+#endif
+
+#undef GET_FBS_ATTR
+#undef GET_DATA_VEC
+
+#define FBS_SET_STR_VAL(OBJ, FUNC, FBS_STR) \
+  if (FBS_STR) {                            \
+    OBJ.FUNC(FBS_STR->str());               \
+  }
+
+Status LoadInitializerOrtFormat(const fbs::Tensor& fbs_tensor,
+                                TensorProto& initializer) {
+  initializer.Clear();
+
+  FBS_SET_STR_VAL(initializer, set_name, fbs_tensor.name());
+  FBS_SET_STR_VAL(initializer, set_doc_string, fbs_tensor.doc_string());
+
+  auto fbs_dims = fbs_tensor.dims();
+  ORT_RETURN_IF_NOT(nullptr != fbs_dims, "fbs_dims cannot be null");
+  initializer.mutable_dims()->Add(fbs_dims->cbegin(), fbs_dims->cend());
+
+  auto fbs_data_type = fbs_tensor.data_type();
+  initializer.set_data_type(fbs_data_type);
+  if (fbs_data_type == fbs::TensorDataType_STRING) {
+    auto fbs_str_data = fbs_tensor.string_data();
+    ORT_RETURN_IF_NOT(nullptr != fbs_str_data, "fbs_str_data cannot be null");
+    auto mutable_str_data = initializer.mutable_string_data();
+    mutable_str_data->Reserve(fbs_str_data->size());
+    for (const auto& fbs_str : *fbs_str_data) {
+      mutable_str_data->Add(fbs_str->str());
+    }
+  } else {
+    auto fbs_raw_data = fbs_tensor.raw_data();
+    // Should we throw in this case? or just leave the raw_data to be empty?
+    ORT_RETURN_IF_NOT(nullptr != fbs_raw_data, "fbs_raw_data cannot be null");
+    // fbs_raw_data is uint8_t vector, so the size is byte size
+    initializer.set_raw_data(fbs_raw_data->Data(), fbs_raw_data->size());
+  }
+
+  return Status::OK();
+}
+
+static Status LoadTensorDimensionOrtFormat(const fbs::Dimension& fbs_dim,
+                                           TensorShapeProto_Dimension& dim) {
+  dim.Clear();
+  FBS_SET_STR_VAL(dim, set_denotation, fbs_dim.denotation());
```

Comment:
nit: it's a static function (so we control the usage) and we only call it immediately after creating TensorShapeProto_Dimension, so the Clear is perhaps unnecessary.